refactor(linter): output smaller spans for unused disable directives with multiple rules

Sysix opened the pull request:
![grafik](https://github.com/user-attachments/assets/4df3b145-7e42-48bf-8d8c-b72aa2b6273e)

camc314 reviewed (approved):
nice!

camc314 commented:
### Merge activity

* **Jun 20, 9:18 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(linter): output smaller spans for unused disable directives with multiple rules (#11781)

![grafik](https://github.com/user-attachments/assets/4df3b145-7e42-48bf-8d8c-b72aa2b6273e)
